x-pack/test/api_integration/apis/ml/results/get_category_examples.ts

```diff
@@ -0,0 +1,198 @@
+      '130.246.123.197 - - [2018-07-22T03:26:21.326Z] "GET /beats/metricbeat_1 HTTP/1.1" 200 6850 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+      '223.87.60.27 - - [2018-07-22T00:39:02.912Z] "GET /elasticsearch/elasticsearch-6.3.2.deb HTTP/1.1" 200 6219 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+    ],
+  };
+
+  async function getCategoryExamples(
+    jobId: string,
+    categoryIds: string[],
+    maxExamples: number,
+    user: USER,
+    expectedStatusCode: number,
+    space?: string
+  ) {
+    const { body } = await supertest
+      .post(`${space ? `/s/${space}` : ''}/api/ml/results/category_examples`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS)
+      .send({ jobId, categoryIds, maxExamples })
+      .expect(expectedStatusCode);
+
+    return body;
+  }
+
+  describe('get category_examples', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/module_sample_logs');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      await ml.api.createAndRunAnomalyDetectionLookbackJob(
+        // @ts-expect-error not full interface
+        testJobConfig,
+        testDatafeedConfig,
+        idSpace1
+      );
+    });
+
+    after(async () => {
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.testResources.deleteIndexPatternByTitle('ft_module_sample_logs');
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should produce the correct 1 example for the job', async () => {
+      const resp = await getCategoryExamples(
+        jobIdSpace1,
+        Object.keys(expectedCategoryExamples1),
+        1,
+        USER.ML_POWERUSER,
+        200,
+        idSpace1
+      );
+
+      expect(resp['1'].length).to.eql(
+        expectedCategoryExamples1['1'].length,
+        `response examples length should be ${expectedCategoryExamples1['1'].length} (got ${resp['1'].length})`
+      );
+    });
+
+    it('should produce the correct 3 examples for the job', async () => {
+      const resp = await getCategoryExamples(
+        jobIdSpace1,
+        Object.keys(expectedCategoryExamples3),
+        3,
+        USER.ML_POWERUSER,
+        200,
+        idSpace1
+      );
+
+      expect(resp['1'].length).to.eql(
+        expectedCategoryExamples3['1'].length,
+        `response examples length should be ${expectedCategoryExamples3['1'].length} (got ${resp['1'].length})`
+      );
+    });
+
+    it('should not produce the correct example for the job', async () => {
```

Comment:
Again, I'm not sure if I understand the test case correctly. It looks like `when we ask for 3 examples of category '2', we don't get results for category '3'  back`. Is that the intended test case?

Reply:
yes, that is the intended test case. 
As with the previous similar test, i don't think this is a useful test and will remove it.

Reply:
Updated in https://github.com/elastic/kibana/pull/124710/commits/7634363c0ce8420f1ce649f17012731ecce30cae